interfaces/hotplug/deviceinfo.go

```diff
@@ -89,3 +90,47 @@ func (h *HotplugDeviceInfo) Attribute(name string) (string, bool) {
 	val, ok := h.Data[name]
 	return val, ok
 }
+
+func (h *HotplugDeviceInfo) firstAttrValueOf(tryAttrs ...string) string {
+	for _, attr := range tryAttrs {
+		if val, _ := h.Attribute(attr); val != "" {
+			return val
+		}
+	}
+	return ""
+}
+
+func (h *HotplugDeviceInfo) String() string {
+	var str []string
+
+	// devname is the name of the device under /dev, eg. /dev/ttyS0;
+	// prefer devname over devpath as this is the one used to talk to the device.
+	if devname := h.DeviceName(); devname != "" {
+		str = append(str, fmt.Sprintf("devname:%s", devname))
+	} else {
+		// devpath is the path of the device under /sys, eg. /sys/devices/pnp0/00:04/tty/ttyS0
+		if devpath := h.DevicePath(); devpath != "" {
+			str = append(str, fmt.Sprintf("devpath:%s", devpath))
+		}
+	}
+	for _, attr := range []string{"MAJOR", "MINOR"} {
```

Comment:
what are these? the device major/minor, they don't seem super useful in this context

Reply:
I was mostly thinking of devs here, but yeah, these are probably not useful after all. Will remove.